src/Symfony/Component/Yaml/Parser.php

```diff
@@ -234,9 +234,11 @@ public function parse($value, $exceptionOnInvalidType = false, $objectSupport =
                     throw new ParseException('Multiple documents are not supported.');
                 }
 
-                // 1-liner optionally followed by newline
-                $lineCount = count($this->lines);
-                if (1 === $lineCount || (2 === $lineCount && empty($this->lines[1]))) {
+                // store the last PCRE regex error code
+                $pregLastError = preg_last_error();
+
+                // 1-liner optionally followed by newline(s)
+                if (preg_match('#^.+\s*#i', $value)) {
```

Comment:
I tried to do that in Parser::cleanup() but that broke things... but doing it here might work